packages/tupaia-web/src/features/DashboardItem/DashboardItem.tsx

```diff
@@ -79,12 +79,14 @@ export const DashboardItem = ({ dashboardItem }: { dashboardItem: DashboardItemT
     viewType,
     name,
   } = config as DashboardItemConfig;
+
   const isExpandable = !!(
     periodGranularity ||
     type === 'chart' ||
     type === 'matrix' ||
     viewType === 'dataDownload' ||
-    viewType === 'filesDownload'
+    viewType === 'filesDownload' ||
+    (viewType === 'qrCodeVisual' && report && report.data && report?.data?.length > 1)
```

Comment:
Based on team tech design on this we're thinking of doing the following:
- Create an `ExpandableVisualWrapper` component that we explicitly wrap any visual that supports expanding
- Move the logic of rendering the `ExpandedVisualModal` into that wrapper
- Create a `useIsExpanded()` hook to check if the viz is expanded rather than passing around an `isExpanded` prop in `tupaia-web`